test/tap/doctor.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+var test = require('tap').test
+var common = require('../common-tap.js')
+var npm = require('../../lib/npm.js')
+var http = require('http')
+var url = ''
+
+test('setup', function (t) {
+  http.createServer(function (q, s) {
+    s.end(JSON.stringify([{lts: true, version: '0.0.0'}]))
+    this.close()
+  }).listen(common.port, function () {
+    // Fixture
+    url = 'http://localhost:' + common.port
+    npm.load({}, function (err) {
+      t.ifError(err, 'npm loaded successfully')
+      t.pass('all set up')
+      t.done()
+    })
+  })
+})
+
+test('npm doctor', function (t) {
+  npm.commands.doctor({url: 'http://localhost:1337/'}, true, function (err, list) {
+    t.ifError(err, 'npm loaded successfully')
+    t.same(list.length, 5, 'list should have 5 prop')
```

Comment:
Yeah it should be since `npm doctor` ask the things to the registry through `npm ping` and `fetchPackageMetadata`. I've added `npm-registry-mock` into this test and verifying all of the output!